iibb.py

```diff
@@ -122,29 +123,27 @@ def ConsultarContribuyentes(self, fecha_desde, fecha_hasta, cuit_contribuyente):
             self.xml.contribuyentes.contribuyente.cuitContribuyente = cuit_contribuyente
 
             xml = self.xml.as_xml()
-            self.CodigoHash = md5.md5(xml).hexdigest()
+            self.CodigoHash = md5(xml.encode('utf-8')).hexdigest()
             nombre = "DFEServicioConsulta_%s.xml" % self.CodigoHash
 
             # guardo el xml en el archivo a enviar y luego lo re-abro:
-            archivo = open(os.path.join(tempfile.gettempdir(), nombre), "w")
-            archivo.write(xml)
-            archivo.close()
-            archivo = open(os.path.join(tempfile.gettempdir(), nombre), "r")
+            with open(os.path.join(tempfile.gettempdir(), nombre), "w") as archivo:
+                archivo.write(xml)
 
             if not self.testing:
-                response = self.client(
-                    user=self.Usuario, password=self.Password, file=archivo
-                )
+                with open(os.path.join(tempfile.gettempdir(), nombre), "r") as archivo:
+                    response = self.client(
+                        user=self.Usuario, password=self.Password, file=archivo)
             else:
-                response = open(self.testing).read()
+                with open(self.testing).read() as archivo:
```

Comment:
Para el context manager, el `read` debería estar en el archivo, ¿o lo armaron asi por algún motivo en especial?:
```suggestion
                with open(self.testing) as archivo:
```